HDF5 doesn't build on Expanse

```1 checking for vsnprintf... yes
  2 checking for waitpid... yes
  3 checking for roundf... yes
  4 checking for lroundf... yes
  5 checking for llroundf... yes
  6 checking for round... yes
  7 checking for lround... yes
  8 checking for llround... yes
  9 checking for an ANSI C-conforming const... yes
 10 checking if the compiler understands  __inline__... yes
 11 checking if the compiler understands __inline... yes
 12 checking if the compiler understands inline... yes
 13 checking for __attribute__ extension... yes
 14 checking for __func__ extension... yes
 15 checking for __FUNCTION__ extension... yes
 16 checking for C99 designated initialization support... yes
 17 checking how to print long long... %ld and %lu
 18 checking enable debugging symbols... yes
 19 checking enable asserts... yes
 20 checking enable developer warnings... no
 21 checking profiling... no
 22 checking optimization level... debug
 23 checking for internal debug output... AC,B2,CX,D,F,HL,I,O,S,ST,T,Z
 24 checking whether function stack tracking is enabled... no
 25 checking for API tracing... yes
 26 checking whether a memory checking tool will be used... no
 27 checking whether internal memory allocation sanity checking is used... yes
 28 checking for parallel support files... provided by compiler
 29 checking whether a simple MPI-IO C program can be linked... no
 30 configure: error: unable to link a simple MPI-IO C program
```

```
module list 1) shared   2) slurm/expanse/20.02.3   3) DefaultModules   4) cpu/0.15.4   5) gcc/9.2.0   6) cmake/3.18.2   7) openmpi/3.1.6
```

--- a/dependencies/_install.sh
+++ b/dependencies/_install.sh
@@ -117,7 +117,7 @@ log_filepath=$log_dir"/HDF5.log"
 
 cd $src_dir"/HDF5"
     echo "|  |--> Configuring (./configure)..."
-    ./configure --prefix=$build_dir --enable-parallel --enable-deprecated-symbols > $log_filepath 2>&1
+    ./configure CC=mpicc CXX=mpicxx --prefix=$build_dir --enable-parallel --enable-deprecated-symbols > $log_filepath 2>&1
 
     echo "|  |--> Building (w/ Make)..."
     make "$@" >> $log_filepath 2>&1

